libshortfin/src/shortfin/local/messaging.h

```diff
@@ -140,7 +140,7 @@ class SHORTFIN_API Message {
 };
 
 // Future specialization for Message::Ref.
-extern template class TypedFuture<Message::Ref>;
+template class TypedFuture<Message::Ref>;
```

Comment:
This is intentional. Does MSVC not support extern templates?

Reply:
https://learn.microsoft.com/en-us/cpp/error-messages/compiler-warnings/compiler-warning-level-1-c4910?view=msvc-170`
> The explicit template instantiation named _<identifier>_ is modified by both the `__declspec(dllexport)` and `extern` keywords. However, these keywords are mutually exclusive. The `__declspec(dllexport)` keyword means instantiate the template class, while the `extern` keyword means do not automatically instantiate the template class.

These two types were already declared using `SHORTFIN_API`, i.e. `__declspec(dllexport)`:

https://github.com/nod-ai/SHARK-Platform/blob/61eacac59317f2b2db1e54778ca8a850db6f2734/libshortfin/src/shortfin/local/async.h#L160-L162 https://github.com/nod-ai/SHARK-Platform/blob/61eacac59317f2b2db1e54778ca8a850db6f2734/libshortfin/src/shortfin/support/api.h#L13-L27